### PR TITLE
Subjects tei

### PR DESCRIPTION
**Pull request by alicefuzier**
For #1970

### CODE REVIEW

File: pipeline/transformer/transformer_tei/src/main/scala/weco/pipeline/transformer/tei/transformers/TeiSubjects.scala [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        val label = NormaliseText(term.text)
+        val reference = parseReference(term)
+        val id = (reference, identifierType) match {
+          case (Some(r), Some(identifierType)) =>
+            Identifiable(SourceIdentifier(identifierType, "Subject", r))
+          case _ => Unidentifiable
+        }
+        label.map(l =>
+          Subject(id = id, label = l, concepts = List(Concept(label = l))))
+      }
```
Comment (alexwlchan) [inline pipeline/transformer/transformer_tei/src/main/scala/weco/pipeline/transformer/tei/transformers/TeiSubjects.scala:51]: Really easy to follow. 👍

File: pipeline/transformer/transformer_tei/src/main/scala/weco/pipeline/transformer/tei/transformers/TeiSubjects.scala [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    }.toList
+
+  private def parseReference(term: Node) = {
+    val referenceString = term \@ "ref"
+    // arabic manuscripts seem to have the subject id in the
+    // attribute "key" instead of "ref" ¯\_(ツ)_/¯
+    val idValue =
+      if (referenceString.isEmpty) term \@ "key" else referenceString
+    // some of the subject ids are prepended with "subject_sh" for lcsh or " subject" for mesh.
+    // So here we remove that to get the id
```
Comment (alexwlchan) [inline pipeline/transformer/transformer_tei/src/main/scala/weco/pipeline/transformer/tei/transformers/TeiSubjects.scala:61]: Is the 'sh' part of the LCSH identifier, or do we want to bin that too? It's not obvious to me.

e.g. in your example is the identifier `sh85083116` or `85083116`

**Review by alexwlchan — APPROVED** [review]
Super nice. ✨ 

One question about LCSH identifiers but otherwise this can go in.

File: pipeline/transformer/transformer_tei/src/main/scala/weco/pipeline/transformer/tei/transformers/TeiSubjects.scala (hunk 2 above)
Comment (alicefuzier) [reply pipeline/transformer/transformer_tei/src/main/scala/weco/pipeline/transformer/tei/transformers/TeiSubjects.scala]: Yeah that comment was written poorly, I've updated it.
Yes, for lcsh `sh` is part of the id so we keep it , we only discard the `subject_` bit before

**Pull request merged**